yt/data_objects/data_containers.py

```diff
@@ -662,6 +663,182 @@ def to_glue(self, fields, label="yt", data_collection=None):
+
+                >>> ramses_ds = yt.load(
+                ...     "/Users/agurvich/Desktop/yt_workshop/"+
+                ...     "DICEGalaxyDisk_nonCosmological/output_00002/info_00002.txt")
+
+                >>> region = ramses_ds.sphere(ramses_ds.domain_center,(1000,'kpc'))
+
+                >>> reader = region.create_firefly_object(
+                ...     path_to_firefly="/Users/agurvich/research/repos/Firefly",
+                ...     fields_to_include=[
+                ...     'particle_extra_field_1',
+                ...     'particle_extra_field_2'],
+                ...     fields_units = ['dimensionless','dimensionless'],
+                ...     dataset_name = 'IsoGalaxyRamses')
+
+                >>> reader.options['color']['io']=[1,1,0,1]
+                >>> reader.particleGroups[0].decimation_factor=100
+                >>> reader.dumpToJSON()
+            """ 
+        
+        ## attempt to import firefly_api
+        try:
+            from firefly_api.reader import Reader
+            from firefly_api.particlegroup import ParticleGroup
+            from firefly_api.errors import FireflyError,FireflyWarning
+            from firefly_api.errors import warnings as firefly_warnings
+        except ImportError:
+            raise ImportError("Can't find firefly_api, ensure it"+
+                "is in your python path or install it with"+
+                "'$ pip install firefly_api'. It is also available"+
+                "on github at github.com/agurvich/firefly_api")
+        try:
+            assert len(fields_units) == len(fields_to_include)
+        except AssertionError:
+            raise FireflyError("Each requested field must have units.")
+        
+        ## for safety, in case someone passes a float just cast it
+        default_decimation_factor = int(default_decimation_factor)
+        
+        ## initialize a firefly reader instance
+        reader = Reader(
+            JSONdir=os.path.join(path_to_firefly,'data',dataset_name),
+            prefix='ytData',
+            clean_JSONdir=True,
+            )
+
+        ## create a ParticleGroup object that contains *every* field
+        for ptype in sorted(self.ds.particle_types_raw):
+            ## skip this particle type if it has no particles in this dataset 
+            if self[ptype,'relative_particle_position'].shape[0]==0:
+                continue
+            
+            if show_unused_fields:
+                ## read the available extra fields from yt
+                this_ptype_fields = self.ds.particle_fields_by_type[ptype]
+
+                ## load the extra fields
+                for field in this_ptype_fields:
+                    if 'position' in field or 'velocity' in field:
+                        continue
+                    if field not in fields_to_include:
+                        firefly_warnings.warn(FireflyWarning(
+                            'detected (but did not request) {} {}'.format(ptype,field)))
```

Comment:
Here I'd use e.g. `mylog.warn` so the warnings get printed in yt's log output like other yt warning messages.

Reply:
I see, I didn't realize that's how that was done.